### ISSUE TITLE
Sometimes the fitter give internal error or unknown cell: id00001 - hints on root cause?

### ISSUE THREAD

**Issue by MattisLind** [hoglet67/atf15xx_yosys#4]
Hello!

This is not a bug report for this project, rather a question regarding the Atmel fitters which you might have some input on since you probably have been using them a lot more that I have.

I have fiddled with some code that I tried to fit but I keep getting either of these two errors:

```
Atmel ATF1508 Fitter Version 1918 (3-21-07)
Copyright 1999,2000 Atmel Corporation


INTERNAL ERROR - Please contact your Hot-Line
```

```
Atmel ATF1508 Fitter Version 1918 (3-21-07)
Copyright 1999,2000 Atmel Corporation
Unknown cell : id00001
Reading input file failed ...
```

I have tried to narrow it down by creating a minimal case that provokes this but has been unsuccessful this far. All the minimal cases are fitted successfully. It is when I am close to the full design this happens. I have partioned the design in separate chunks that individually fit into a 1508 and I cannot find that the sum of the consumed resources would exceed the total amount of available resources. 

Have you seen this type of errors? Did you find the root cause for these kind of errors?

**Comment by hoglet67** [comment]
> Have you seen this type of errors? Did you find the root cause for these kind of errors?

Yes I have, and the cause is usually something unexpected in the EDIF file.

If possible, could you attach:
- your verilog source
- the .edif file
- and a log of yosys running

**Comment by hoglet67** [comment]
You just need to look in the EDIF to see what id00001 is.

In this case it's:
```
    (cell (rename id00001 "$_TBUF_")
      (cellType GENERIC)
      (view VIEW_NETLIST
        (viewType NETLIST)
        (interface
          (port A (direction INPUT))
          (port E (direction INPUT))
          (port Y (direction OUTPUT))
        )
      )
    )
```
This should be mapped to a TRI cell by the -toutpad option in the iopadmap command in the yosys script:
```
iopadmap -bits -inpad INBUF Q:A -outpad BUF A:Q -toutpad TRI ENA:A:Q -tinoutpad bibuf EN:Q:A:PAD
```
For some reason in your case this isn't working.,

I wonder if it's because the in the VHDL you have defined the port as inout, but it's actually only being used as an out?

Does changing the data port to from inout to out help?

**Comment by MattisLind** [comment]
I just tried to change it into "out" only and there is no difference. Still same error message.

[minimalcase.edif.txt](https://github.com/hoglet67/atf15xx_yosys/files/14923446/minimalcase.edif.txt)

**Comment by hoglet67** [comment]
OK, leave this with me and I'll try to work out what's wrong. It may be a few days...

It would be interesting to see the same test case in Verilog has the same issue.

**Comment by hoglet67** [comment]
> Ah, I think I may have found the problem. I saved the above code as `Add.v` and this works fine. Once I change the filename to, e.g. `fred.v`, then I see the `Unknown cell : id00001` issue.

That makes sense.... The run_yosys script was inferring the top level module from the verilog filename. I've updated this to use the -auto-top parameter, so it shouldn't now matter what the filename is.

Hopefully -auto-top does a good job of identifying the top of the design hierarchy. I guess we'll find out over time.
